From f0f9db538cb0d951847b885f82e930bb39bfd733 Mon Sep 17 00:00:00 2001
From: Saikrishna Edupuganti <saikrishna.edupuganti@intel.com>
Date: Fri, 28 Jun 2019 07:23:35 +0000
Subject: [PATCH] include libbpf

Signed-off-by: Saikrishna Edupuganti <saikrishna.edupuganti@intel.com>
---
 core/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 804ff9e7..8ce084c3 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -78,9 +78,9 @@ endif
 
 # We always want these libraries to be dynamically linked even when the
 # user requests a static build.
-ALWAYS_DYN_LIBS := -lpthread -ldl
+ALWAYS_DYN_LIBS := -lpthread -ldl -lelf -lz
 # These libraries are not supported by pkg-config.
-ALWAYS_LIBS := -lpcap -lgflags -lnuma
+ALWAYS_LIBS := -lpcap -lgflags -lnuma -lbpf
 # If pkg-config is available, we just need a list of the dependecies.
 PKG_CONFIG_DEPS := libglog protobuf grpc++ libunwind zlib
 # If pkg-config is not available, we need to list the libs we depend on.
-- 
2.17.1

