From 08652fd107b01a3e5d8e434ab62c3dbd2aa93d63 Mon Sep 17 00:00:00 2001
From: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>
Date: Tue, 17 Mar 2020 13:39:46 -0700
Subject: [PATCH] legacy mem

Signed-off-by: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>
---
 core/dpdk.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/dpdk.cc b/core/dpdk.cc
index c2a220cb..17d6687b 100644
--- a/core/dpdk.cc
+++ b/core/dpdk.cc
@@ -108,6 +108,7 @@ void init_eal(int dpdk_mb_per_socket, std::string nonworker_corelist) {
       // Do not bother with /var/run/.rte_config and .rte_hugepage_info,
       // since we don't want to interfere with other DPDK applications.
       "--no-shconf",
+      "--legacy-mem",
   };
 
   if (dpdk_mb_per_socket <= 0) {
-- 
2.25.1

