From 70f44d0724e02b751e89a8763b073929384c76ab Mon Sep 17 00:00:00 2001
From: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>
Date: Fri, 14 Feb 2020 15:02:10 -0800
Subject: [PATCH] More RTE-specific macro update.

Signed-off-by: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>
---
 core/drivers/pmd.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/drivers/pmd.cc b/core/drivers/pmd.cc
index 46b86fe..2ceb0a1 100644
--- a/core/drivers/pmd.cc
+++ b/core/drivers/pmd.cc
@@ -408,9 +408,9 @@ CommandResponse PMDPort::UpdateConf(const Conf &conf) {
   rte_eth_dev_stop(dpdk_port_id_);
 
   if (conf_.mtu != conf.mtu && conf.mtu != 0) {
-    if (conf.mtu > SNBUF_DATA || conf.mtu < ETHER_MIN_MTU) {
+    if (conf.mtu > SNBUF_DATA || conf.mtu < RTE_ETHER_MIN_MTU) {
       return CommandFailure(EINVAL, "mtu should be >= %d and <= %d",
-                            ETHER_MIN_MTU, SNBUF_DATA);
+                            RTE_ETHER_MIN_MTU, SNBUF_DATA);
     }
 
     int ret = rte_eth_dev_set_mtu(dpdk_port_id_, conf.mtu);
-- 
2.7.4

