From 953e363359634e3513e770ae87474d6fdb7fdb72 Mon Sep 17 00:00:00 2001
From: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>
Date: Tue, 28 Apr 2020 10:43:05 -0700
Subject: [PATCH] Adding arguments for IPChecksum and L4Checksum.

Signed-off-by: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>
---
 protobuf/module_msg.proto | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/protobuf/module_msg.proto b/protobuf/module_msg.proto
index 716b3870..2a1b1f15 100644
--- a/protobuf/module_msg.proto
+++ b/protobuf/module_msg.proto
@@ -982,6 +982,30 @@ message SourceArg {
   uint64 pkt_size = 1; /// The size (in bytes) of packet data to produce.
 }
 
+/**
+* The IPChecksum module calculates the IPv4 checksum of packets. If
+* verify is set to true, the module can be used to validate the checksum
+* of the IPv4 packet
+*
+* __Input Gates__: 1
+* __Output Gates__: 2
+*/
+message IPChecksumArg {
+ bool verify = 1; /// check checksum
+}
+
+/**
+* The L4Checksum module calculates the UDP/IPv4 checksum of packets. If
+* verify is set to true, the module can be used to validate the checksum
+* of the UDP/IPv4 packet
+*
+* __Input Gates__: MAX_GATES
+* __Output Gates__: 2
+*/
+message L4ChecksumArg {
+ bool verify = 1; /// check checksum
+}
+
 /**
  * The GtpuEcho module processes the GTPv1 echo packet and prepares
  * corresponding IP packet containing GTP echo response. It assumes
-- 
2.25.1

