From df5f5abf5806ab51daba6582c20c34e8d709277a Mon Sep 17 00:00:00 2001
From: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>
Date: Wed, 27 May 2020 21:04:19 -0700
Subject: [PATCH] Bypass l4/ip4 csum compute/validate for !tcp/!udp

Signed-off-by: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>
Signed-off-by: Saikrishna Edupuganti <saikrishna.edupuganti@intel.com>
Signed-off-by: Hyunsun Moon <hyunsun@opennetworking.org>
---
 core/modules/l4_checksum.cc | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/core/modules/l4_checksum.cc b/core/modules/l4_checksum.cc
index 3ad1f23d..5236a1c3 100644
--- a/core/modules/l4_checksum.cc
+++ b/core/modules/l4_checksum.cc
@@ -76,8 +76,12 @@ void L4Checksum::ProcessBatch(Context *ctx, bess::PacketBatch *batch) {
       if (verify_)
 	EmitPacket(ctx, batch->pkts()[i],
 		   (VerifyIpv4TcpChecksum(*ip, *tcp)) ? FORWARD_GATE : FAIL_GATE);
-      else
-	tcp->checksum = CalculateIpv4TcpChecksum(*ip, *tcp);
+      else {
+	tcp->checksum = CalculateIpv4TcpChecksum(*ip, *tcp);	
+	EmitPacket(ctx, batch->pkts()[i], FORWARD_GATE);
+      }
+    } else { /* fail-safe condition. */
+	EmitPacket(ctx, batch->pkts()[i], FORWARD_GATE);
     }
   }
 }
-- 
2.25.1

