From 01f457b5d29a64b5ba434da3009c8b3200ca9a2c Mon Sep 17 00:00:00 2001
From: Aaron Sinclair <aaron@selfienetworks.com>
Date: Wed, 17 Jun 2020 09:59:53 -0700
Subject: [PATCH] Include cjose and jansson for NTF

---
 core/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/Makefile b/core/Makefile
index b2921e10..6252b141 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -80,7 +80,7 @@ endif
 # user requests a static build.
 ALWAYS_DYN_LIBS := -lpthread -ldl
 # These libraries are not supported by pkg-config.
-ALWAYS_LIBS := -lpcap -lgflags -lnuma -lbpf
+ALWAYS_LIBS := -lpcap -lgflags -lnuma -lbpf -lcjose -ljansson -lcrypto
 # If pkg-config is available, we just need a list of the dependecies.
 PKG_CONFIG_DEPS := libglog protobuf grpc++ libunwind zlib
 # If pkg-config is not available, we need to list the libs we depend on.
-- 
2.24.1

