From 4b459fb183b66517ae724fa37b4344af4cda8851 Mon Sep 17 00:00:00 2001
From: Amar Sri <amarsri28@gmail.com>
Date: Thu, 12 Aug 2021 03:16:10 -0700
Subject: [PATCH] Protobuf changes for Qos.cc

---
 protobuf/module_msg.proto | 35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/protobuf/module_msg.proto b/protobuf/module_msg.proto
index 25dfc81e..1f7058cd 100644
--- a/protobuf/module_msg.proto
+++ b/protobuf/module_msg.proto
@@ -1291,3 +1291,41 @@ message MplsPopArg {
 message WorkerSplitArg {
   map<uint32, uint32> worker_gates = 1; // ogate -> worker mask
 }
+
+message QosArg {
+  repeated Field fields = 1;
+  repeated Field values = 2;
+}
+
+message QosCommandAddArg {
+  uint64 gate = 1;
+  uint64 cir = 2;
+  uint64 pir = 3;
+  uint64 cbs = 4;
+  uint64 pbs = 5;
+  uint64 ebs = 6;
+  oneof optional_deduct_len {
+    int64 deduct_len = 9;
+  }
+  repeated FieldData fields = 7;
+  repeated FieldData values = 8;
+}
+
+message QosCommandDeleteArg {
+  repeated FieldData fields = 2;
+}
+
+/**
+ * The function `clear()` for WildcardMatch takes no parameters, it clears
+ * all state in the WildcardMatch module (is equivalent to calling delete for all rules)
+ */
+message QosCommandClearArg {
+}
+
+/**
+ * For traffic which does not match any rule in the WildcardMatch module,
+ * the `set_default_gate(...)` function specifies which gate to send this extra traffic to.
+ */
+message QosCommandSetDefaultGateArg {
+  uint64 gate = 1;
+}
--
2.17.1

