diff --git a/core/dpdk.cc b/core/dpdk.cc
index 964d513c..130572d7 100644
--- a/core/dpdk.cc
+++ b/core/dpdk.cc
@@ -112,7 +112,7 @@ class CmdLineOpts {
 void init_eal(int dpdk_mb_per_socket, std::string nonworker_corelist) {
   CmdLineOpts rte_args{
       "bessd",
-      "--master-lcore",
+      "--main-lcore",
       std::to_string(RTE_MAX_LCORE - 1),
       "--lcore",
       std::to_string(RTE_MAX_LCORE - 1) + "@" + nonworker_corelist,
diff --git a/core/memory_test.cc b/core/memory_test.cc
index 9849147d..c5e44a16 100644
--- a/core/memory_test.cc
+++ b/core/memory_test.cc
@@ -139,9 +139,9 @@ TEST_P(HugepageTest, LeakFree) {
   } while (get_cpu_time() - start < 0.5);  // 0.5 second for each page size
 }
 
-INSTANTIATE_TEST_CASE_P(PageSize, HugepageTest,
-                        ::testing::Values(HugepageSize::k2MB,
-                                          HugepageSize::k1GB));
+INSTANTIATE_TEST_SUITE_P(PageSize, HugepageTest,
+                         ::testing::Values(HugepageSize::k2MB,
+                                           HugepageSize::k1GB));
 
 TEST(DmaMemoryPoolTest, PoolSetup) {
   if (geteuid() != 0) {
