From c86f875d80fed98ce8732066a1d0429e1dba8154 Mon Sep 17 00:00:00 2001
From: Amar Sri <amarsri28@gmail.com>
Date: Fri, 17 Jun 2022 05:49:38 -0700
Subject: [PATCH] pbuf

---
 protobuf/module_msg.proto | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/protobuf/module_msg.proto b/protobuf/module_msg.proto
index c435437a..20bb438a 100644
--- a/protobuf/module_msg.proto
+++ b/protobuf/module_msg.proto
@@ -1332,3 +1332,12 @@ message QosCommandClearArg {
 message QosCommandSetDefaultGateArg {
   uint64 gate = 1;
 }
+
+message SchCommandSetDefaultGateArg {
+  uint64 gate = 1;
+}
+
+message SchArg {
+  repeated Field fields = 1;
+  repeated Field values = 2;
+}
-- 
2.17.1

